packages/itwinui-react/src/core/utils/hooks/useStyles.ts

```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect.js';
+import { getDocument } from '../functions/dom.js';
+import * as React from 'react';
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment -- only gets created at build-time
+// @ts-ignore
+import rawCssText from '../../../styles.js';
+
+// react <18 fallback for useInsertionEffect, with workaround for webpack getting rid of React namespace
+const _React = React;
+const useIsomorphicInsertionEffect =
+  _React.useInsertionEffect ?? useIsomorphicLayoutEffect;
+
+/**
+ * Dynamically loads the iTwinUI styles as a constructed stylesheet,
+ * falling back to a regular `<style>` tag in `<head>`.
+ */
+export const useStyles = (options?: {
+  withLayer?: boolean;
+  document?: () => Document | undefined;
+}) => {
+  const { withLayer = true, document = () => getDocument() } = options ?? {};
+  const loaded = _React.useRef(false);
+
+  useIsomorphicInsertionEffect(() => {
+    if (!loaded.current) {
+      loadStyles({ withLayer, document });
+      loaded.current = true;
+    }
+  }, [document, loaded, withLayer]);
+};
```

Comment:
Does this append the new layer onto the current styling?

Reply:
not sure i understand the question, can you rephrase?

Reply:
I re-read some more of the PR, never mind. Thanks